workflow-templates/im-deploy-iis-website.yml

```diff
@@ -309,54 +305,56 @@ jobs:
+      #     website-host-header: '${{ env.WEBSITE_HOST_HEADER }}'
+      #     website-path: '${{ env.WEBSITE_PATH }}'
+      #     website-cert-path: '${{ env.WEBSITE_CERT_PATH }}'
+      #     website-cert-friendly-name: '${{ env.WEBSITE_CERT_FRIENDLY_NAME }}'
+      #     website-cert-password: '${{ secrets.WEBSITE_CERT_PASSWORD }}'
+      #     app-pool-service-account-id: ${{ env.APP_POOL_SA_ID }}
+      #     app-pool-service-account-secret: ${{ secrets.APP_POOL_SA_SECRET }}
+      #     deployment-service-account-id: ${{ env.DEPLOYMENT_SA_ID }}
+      #     deployment-service-account-secret: ${{ secrets.DEPLOYMENT_SA_SECRET }}
 
       - name: Stop App Pool
         id: app-pool-stop
         if: always()
-        uses: im-open/app-pool-action@v2.0.0
+        uses: im-open/app-pool-action@v2.0.2
         with:
           action: 'stop'
-          server: ${{ needs.set-vars.outputs.IIS_SERVER }}
-          app-pool-name: ${{ needs.set-vars.outputs.APP_POOL_NAME }}
-          service-account-id: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_USER }}
-          service-account-password: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_PASSWORD }}
+          server: ${{ env.IIS_SERVER }}
+          app-pool-name: ${{ env.APP_POOL_NAME }}
+          service-account-id: ${{ env.DEPLOYMENT_SA_ID }}
+          service-account-password: ${{ secrets.DEPLOYMENT_SA_SECRET }}
 
       - name: Deploy deployment package
         if: steps.app-pool-stop.outcome == 'success'
-        uses: 'im-open/deploy-windows-files@v2.0.0'
+        uses: im-open/deploy-windows-files@v2.0.3
         with:
-          server: ${{ needs.set-vars.outputs.IIS_SERVER }}
+          server: ${{ env.IIS_SERVER }}
           source-zip-file-path: ./${{ env.DEPLOY_ZIP }}
-          deployment-folder-path: ${{ needs.set-vars.outputs.WEBSITE_PATH }}
+          deployment-folder-path: ${{ env.WEBSITE_PATH }}
           clean-deployment-folder: 'true'
-          service-account-id: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_USER }}
-          service-account-password: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_PASSWORD }}
+          service-account-id: ${{ env.DEPLOYMENT_SA_ID }}
+          service-account-password: ${{ secrets.DEPLOYMENT_SA_SECRET }}
 
       - name: Start App Pool
-        uses: im-open/app-pool-action@v2.0.0
+        uses: im-open/app-pool-action@v2.0.2
         with:
           action: 'start'
-          server: ${{ needs.set-vars.outputs.IIS_SERVER }}
-          app-pool-name: ${{ needs.set-vars.outputs.APP_POOL_NAME }}
-          service-account-id: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_USER }}
-          service-account-password: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_PASSWORD }}
+          server: ${{ env.IIS_SERVER }}
+          app-pool-name: ${{ env.APP_POOL_NAME }}
+          service-account-id: ${{ env.DEPLOYMENT_SA_ID }}
+          service-account-password: ${{ secrets.DEPLOYMENT_SA_SECRET }}
```

Comment:
You've basically done what Octopus was doing which is fine, but there are [better](https://docs.microsoft.com/en-us/troubleshoot/developer/visualstudio/language-compilers/deploy-aspnet-app-xcopy-command) [ways](https://docs.microsoft.com/en-us/iis/publish/deploying-application-packages/taking-an-application-offline-before-publishing) to deploy to IIS sites.

* copy and unzip the new files to a _new_ directory (its name might include a timestamp or something to ensure it's unique)
* drop an [App_Offline.htm](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/app-offline?view=aspnetcore-6.0) file into the _existing_ directory (may not actually be necessary)
* modify the IIS site's physical directory, changing it to the _new_ directory.
* delete the old directory (or leave one or two versions around for a while and delete older ones)

That process should cause a new hosting process to start with the new code to handle new requests while the old hosting process finishes handling current requests and then shuts down.